app/src/main/java/com/ojasvi/memify/ui/screens/PhotoReasoningScreen.kt

```diff
@@ -73,15 +72,17 @@ fun PhotoReasoningScreen(
 
     PhotoReasoningContents(
         uiState = photoReasoningUiState,
-        onReasonClicked = viewModel::reason
+        onReasonClicked = viewModel::reason,
+        onUpdateImage= viewModel::resetToInitialUiState
     )
 }
 
-@OptIn(ExperimentalComposeUiApi::class, ExperimentalComposeApi::class)
+@OptIn(ExperimentalComposeUiApi::class)
 @Composable
 fun PhotoReasoningContents(
     uiState: PhotoReasoningUiState = PhotoReasoningUiState.Loading,
-    onReasonClicked: (String, Uri) -> Unit = { _, _ -> }
+    onReasonClicked: (String, Uri) -> Unit = { _, _ -> },
```

Comment:
Generally avoid assigning default values to functions. This could be a very hard bug to track.